### ISSUE TITLE
Question about the Global Elements

### ISSUE THREAD

**Issue by kladderadeng** [davideshay/groceries#14]
Is there a way to edit or delete the pre-loaded Global Elements? Having my own global elements could be useful, but it is not necessary. But the opportunity to delete them would be great, since barely and of the preloaded fit our "shopping style", so it would be convenient to edit or delete them.

I suspect I could do this manually in the CouchDB database, but I do not want to break something in future updates.

**Comment by davideshay** [comment]
There is no built-in functionality to delete them. There is no real risk to deleting them in CouchDB and letting that change flow to the frontends.  The only risk would be if you used one of these global items to create a local list item (i.e. in the search bar you selected "Apples" where the app would then create a local copy of the apples item, putting it in your local lists/listgroups, but retaining a reference back to the global item "apples" for translation purposes mainly. But if you deleted the global items before doing anything like that, it would be fine.

If in the future, however, more items are added to the global item pre-populated JSON file in the server app, and the global item "content version" was updated, they would be re-created, along with new content. If you aren't using them, you could just delete them again at that point.

Note, I am working on an Alexa skill that will also have these global items pre-populated, so that would cause some breakage there as well.

**Comment by kladderadeng** [comment]
Thanks for your explanation. In that case it would be safer to just leave the global Items in the list. Would it be possible to exclude the global items from a specific list group maybe, so they are just not shown when picking items in a list / list group? Maybe as an option in the list group like "[x] Exclude Global Items from search" or something like that?

Please don´t take this as something I "demand", just as a humble wish :-)
The issues I have with the global items and why I´d like to get rid of them:

1. Nearly none of them fit our shopping behaviour, so we have to create most item anyway. And now we have a huge list of default items we do not use, which makes picking just more cumbersome than from a curated list.
2. Since I run the UI / App in German, the global ist is not sorted alphabetically. I suppose it is sorted by the original English names of the items, but shown in German.

To sum this up: I am sure the pre-populated global items are very useful for the majority of people, but not for our use. It would make things much easier for us if we could hide them from search, but of course it is probably very specific for us and you would have the work to code it, so I do not expect you to invest the time for such a change of course.

**Comment by davideshay** [comment]
Both issues fixed in 0.7.1. Let me know if there are any other cases of sorting that aren't correctly internationalized yet.
Checks the settings page for the new option to include/exclude global items from the search.

**Comment by kladderadeng** [comment]
Awesome! I absolutely did not expect that to happen at all and now you did it that quick! Thank you very much :-)
I upgraded my Backend, Frontend and the Android App.

I can not confirm (yet) that disabling the global Items works, because unfortunately I deleted them 2 days ago in the DB 🤣 

I had a look at the sorting and in the Web Frontend, the sorting of the Categories looks good. I also tested it in the Android app, but sorting is still off for me. I went to the settings and deleted the local Data, logged in again and hoped that re-loading the data would fix it. But for me it is still not sorted alphabetically on Android:

![image](https://user-images.githubusercontent.com/111387593/235710109-9a5fb60b-8975-4b21-8efd-d932dfe9525c.png)

It´s not a big deal for the categories, it is not a huge list anyway.

**Comment by davideshay** [comment]
Not that you'd want to, but if you go into the DBUUID entry in couchDB, you could change the uomContentVersion back a level/number and restart the backend, and it would re-create the global items.

I cannot replicate the android sorting bug -- everything for me is now sorted correctly on Web and Android. What phone and/or API version are you using? Can you replicate the bug in the Android Emulator? I tend to test with Pixel 5 API 32.

**Comment by kladderadeng** [comment]
Sorry for the delay!

Sorting: It looks to me like this has to do with the language. In the Screenshot above, "Schönheintsprodukte" is something I would translate as "beauty products" which fits in the B category. If this is not reproducable, no worries. The categories are short anyway and this is not really a problem. I do use a Xiaomi Mi9 Lite with Lineage OS 20, which is Android 13. Unfortunately I do not have an Android Emulator. I guess it´s part of an Android Development environment?

DBUUID: Well, my personal opinion usually is to stay as close to the standard as possible. So if changing the DBUUID brings me back to the Standard, I´d be happy to do this. But I am not sure what exactly you mean. I tried to search for it, but I am not smarter.
Looking at the Database with the CouchDB Web UI through the Views and Metadata did not help either. It´s probably easy and I am too dumb to search with the right terms ;-)

**Comment by davideshay** [comment]
The Android Emulator is a part of Android Studio -- a free download that you can fairly simply use to create the Android APK from the source code. You can emulate a variety of Android devices from your development system that way.  That seems like a fairly current build, but is with a "non-Google" OS. It's possible there are some differences in functions that part of the entire Capacitor ecosystem relies on. Not an easy way to test on those.

For the DBUUID, I didn't explain very well. The couchDB server database has one document/record with a "type" of "dbuuid". This contains information about the current database schema version as well as a unique ID to ensure syncing doesn't ever overwrite the wrong database.  

No real need to update anything there if you just install 0.7.3 all of the global items will be recreated as a I created just a few more and also fixed an error on an existing item.

**Comment by kladderadeng** [comment]
Thanks for the explanation!

I can now (that the global Items are back) confirm that the option to hide them works as advertised 👍
